### PR TITLE
CSI-1191: Don't retry when device upgrade is successful but device au…

### PR DESCRIPTION
**Pull request by irshadCypherock**
…th fails

### CODE REVIEW

**Comment by irshadCypherock** [comment]
Task linked: [CSI-1191 BugFix: Handle case when device upgrade is successful but device auth fails on initial](https://app.clickup.com/t/37308523/CSI-1191)

File: cysync/src/pages/mainApp/deviceUpdater/initialFlow/stepperFormComponents/upgradingDevice.tsx [hunk 1]
```diff
@@ -151,6 +162,11 @@ const UpgradingDevice: React.FC<StepComponentProps> = ({ handleClose }) => {
       setTimeout(onClose, 350);
     }
   }, [isCompleted]);
+
+  const isAuthFailed =
+    errorResolutionState ===
+    DeviceUpgradeErrorResolutionState.DEVICE_AUTH_REQUIRED;
```
Comment (HarishRocks) [inline cysync/src/pages/mainApp/deviceUpdater/initialFlow/stepperFormComponents/upgradingDevice.tsx:168]: Wont declaring this check as a variable be a problem as its dependent on a state?  
If we're sure its not a problem, reuse it above on the `refreshComponent` function

**Review by HarishRocks — CHANGES_REQUESTED** [review]

File: cysync/src/pages/initialApp/deviceFlow/deviceSetup/stepperFormComponents/upgradingDevice.tsx [hunk 2]
```diff
@@ -89,11 +92,19 @@ const UpgradingDevice: React.FC<StepComponentProps> = ({
...
   } = useDeviceUpgrade(true);
 
   const refreshComponent = () => {
-    handleRetry();
+    if (
+      errorResolutionState ===
+      DeviceUpgradeErrorResolutionState.DEVICE_AUTH_REQUIRED
+    ) {
+      onClose();
+    } else {
```
Comment (HarishRocks) [inline cysync/src/pages/initialApp/deviceFlow/deviceSetup/stepperFormComponents/upgradingDevice.tsx:105]: Don't do this, if Cysync installs a malicious firmware and auth fails in which case the device is compromised. Cysync will move to main menu.

**Review by HarishRocks — CHANGES_REQUESTED** [review]

File: cysync/src/pages/initialApp/deviceFlow/deviceSetup/stepperFormComponents/upgradingDevice.tsx (hunk 2 above)
Comment (HarishRocks) [reply cysync/src/pages/initialApp/deviceFlow/deviceSetup/stepperFormComponents/upgradingDevice.tsx:105]: lets leave it as it is as we are not restricting the user if they have malicious firmware.

**Review by HarishRocks — APPROVED** [review]

**Pull request merged**